opencodeblocks/blocks/codeblock.py

```diff
@@ -120,11 +159,247 @@ def run_code(self):
+            else:
+                elem.run_color = 0
+
+        QApplication.processEvents()
+        self.transmitting_queue.pop(0)
+        if len(self.transmitting_queue) != 0:
+            # If the queue is not empty, move forward in the animation
+            self.transmitting_animation_in()
+        else:
+            # Else, run the blocks in the self.blocks_to_run
+            self.run_blocks()
+
+    def custom_bfs(self, start_node, reverse=False):
+        """
+        Graph traversal in BFS to find the blocks that are connected to the start_node
+
+        Args:
+            start_node (Block): The block to start the traversal from
+            reverse (bool): If True, traverse in the direction of outputs
+
+        Returns:
+            list: Blocks to run in topological order (reversed)
+            list: each element is a list of blocks/edges to animate in order
+        """
+        # Blocks to run in topological order
+        blocks_to_run = []
+        # List of lists of blocks/edges to animate in order
+        to_transmit = [[start_node]]
+
+        to_visit = [start_node]
+        while len(to_visit) != 0:
+            # Remove duplicates
+            to_visit = list(set(to_visit))
+
+            # Gather connected edges
+            edges_to_visit = []
+            for block in to_visit:
+                blocks_to_run.append(block)
+                if not reverse:
+                    for input_socket in block.sockets_in:
+                        for edge in input_socket.edges:
+                            edges_to_visit.append(edge)
+                else:
+                    for output_socket in block.sockets_out:
+                        for edge in output_socket.edges:
+                            edges_to_visit.append(edge)
+            to_transmit.append(edges_to_visit)
+
+            # Gather connected blocks
+            to_visit = []
+            for edge in edges_to_visit:
+                if not reverse:
+                    to_visit.append(edge.source_socket.block)
+                else:
+                    to_visit.append(edge.destination_socket.block)
+            to_transmit.append(to_visit)
+
+        # Remove start node
+        blocks_to_run.pop(0)
+
+        return blocks_to_run, to_transmit
+
+    def custom_djikstra(self):
+        """
+        Custom graph traversal utility
+        Returns blocks/edges that will potentially be run by run_right
+        from closest to farthest from self
+
+        Returns:
+            list: each element is a list of blocks/edges to animate in order
+        """
+        # Result
+        to_transmit = [[self]]
+
+        # To check if a block has been visited
+        visited = []
+        # We need to visit the inputs of these blocks
+        to_visit_input = [self]
+        # We need to visit the outputs of these blocks
+        to_visit_output = [self]
+
+        # Next stage to put in to_transmit
+        next_edges = []
+        next_blocks = []
+
```

Comment:
Is this the dijkstra path finding algorithm ? It does not look like a djikstra, there is no priority queue or any distance array. 
Moreover, we are working on an acyclic directed graph. The name might be misleading.

Also, the `while` block could be split into 2 `while` blocks for clarity (one for output and one for input)

Reply:
Indeed it is quite far from djikstra, I'll change the name.

I'm not sure what you mean by the two whiles, I don't see a way of doing this